File: api/lib/api/data/task.ex
defmodule Api.Task do
  use Ecto.Schema
  import Ecto.Changeset

  schema "task" do
    field :description, :string
    field :number, :string
    field :parent_org, :integer
    field :slot_weight, :integer
    field :subject, :string
    field :task_id, :integer 
    field :title, :string
  end

  @doc false
  def changeset(task, attrs) do
    task
    |> cast(attrs, [:task_id, :parent_org, :subject, :number, :slot_weight, :title, :description])
    |> validate_required([:task_id, :parent_org, :subject, :number, :slot_weight])
  end
end

File: api_v2/src/main/scala/models/Task.scala
package models

case class Task(
  task_id: Int,
  parent_org: Int,
  subject: String,
  number: String,
  slot_weight: Option[Int],
  title: Option[String],
  description: Option[String]
) extends Executable {
  override val executable_id: Int = task_id
}
